### PR TITLE
4.0.0 - Migrate to async/await

### LINKED ISSUE #13: Remove express as a dependency
I must've just missed this, but there's no reason to include express as a dependeny. Sure people will use it for express, but by removing express we reduce our dependency map on our side

### LINKED ISSUE #23: Errors problems with @node-oauth/express-oauth-server and @node-oauth/oauth2-server version 5
After the recent update to the `@node-oauth/oauth2-server` package to version `5.0.x`, there is a bug related to error evaluation. 

The main issue arises when an error is thrown from `@node-oauth/express-oauth-server`, which still uses `@node-oauth/oauth2-server` version `4.3.0`. Due to this version discrepancy, the error instances from the older version do not match those from the new version `5.0.x`, leading to compatibility issues.

Just pointing out the issue as there is already a [PR](https://github.com/node-oauth/express-oauth-server/pull/22) which updates the version of `@node-oauth/oauth2-server` to the latest one.

### PR DESCRIPTION
**Pull request by jankapunkt**
Might be breaking for older setups!

- bump minimal node to 16
- upgrade @node-oauth/oauth2-server to 5.1.0
- drop bluebird dependency
- upgrade all deps / dev-deps
- refactor all code to minimum es6
- use native async/await
- update ci to use minimum node 16
- update readme and changelog
- make express peer depdendency with `*` value so there will always be the express version used that is installed by users
- test coverage to 100%

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,144 +1,166 @@
...
+
+/**
+ * Grant Middleware.
+ *
+ * Returns middleware that will grant tokens to valid requests.
+ *
+ * (See: https://tools.ietf.org/html/rfc6749#section-3.2)
+ */
+
+ExpressOAuthServer.prototype.token = function(options) {
```
Comment (jorenvandeweyer) [inline index.js]: Why isnt this a class method?

**Review by jorenvandeweyer — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (jankapunkt) [reply index.js]: Oh, you're right they should all be an es6 class. I did this once but reverted the changes and forgot to write as class, again then. Will do.

**Comment by jankapunkt** [comment]
@jorenvandeweyer please note that `express` is now a peer dependency, since there will soon be a major transition to express 5 while many ppl will still run on express 4 and from what I see this peer dependency is a good way to avoid major version conflicts by always pointing to the user's express installation. On the tests we use latest express 4 rn, though

**Comment by jankapunkt** [comment]
@jorenvandeweyer resolved the issues, can you please review, again?

**Review by jorenvandeweyer — APPROVED** [review]

**Comment by jankapunkt** [comment]
Published `@node-oauth/express-oauth-server@4.0.0-rc.0`

**Pull request merged**